src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -39,47 +101,87 @@ out ImmutableArray<(BoundExpression, BoundDagTemp)> bindings)
         }
 
-        /// <summary>
-        /// Used to create a decision dag for a switch statement.
-        /// </summary>
-        /// <param name="syntax"></param>
-        /// <param name="switchExpression"></param>
-        /// <param name="switchSections"></param>
-        /// <param name="defaultLabel"></param>
-        /// <returns></returns>
-        public BoundDecisionDag CreateDecisionDag(
+        private BoundDecisionDag CreateDecisionDag(
             SyntaxNode syntax,
-            BoundExpression switchExpression,
-            ImmutableArray<BoundPatternSwitchSection> switchSections,
-            LabelSymbol defaultLabel)
+            BoundExpression switchGoverningExpression,
+            ImmutableArray<BoundPatternSwitchSection> switchSections)
         {
-            ImmutableArray<PartialCaseDecision> cases = MakeCases(switchExpression, switchSections);
-            BoundDecisionDag dag = MakeDecisionDag(syntax, cases, defaultLabel);
+            ImmutableArray<PartialCaseDecision> cases = MakeCases(switchGoverningExpression, switchSections);
+            BoundDecisionDag dag = MakeDecisionDag(syntax, cases);
             return dag;
         }
 
-        private ImmutableArray<PartialCaseDecision> MakeCases(BoundExpression switchExpression, ImmutableArray<BoundPatternSwitchSection> switchSections)
+        private ImmutableArray<PartialCaseDecision> MakeCases(BoundExpression switchGoverningExpression, ImmutableArray<BoundPatternSwitchSection> switchSections)
         {
-            var rootIdentifier = new BoundDagTemp(switchExpression.Syntax, switchExpression.Type, null, 0);
+            var rootIdentifier = new BoundDagTemp(switchGoverningExpression.Syntax, switchGoverningExpression.Type, null, 0);
             int i = 0;
             var builder = ArrayBuilder<PartialCaseDecision>.GetInstance(switchSections.Length);
             foreach (BoundPatternSwitchSection section in switchSections)
             {
                 foreach (BoundPatternSwitchLabel label in section.SwitchLabels)
                 {
-                    builder.Add(MakePartialCaseDecision(++i, rootIdentifier, label));
+                    if (label.Syntax.Kind() != SyntaxKind.DefaultSwitchLabel)
+                    {
+                        builder.Add(MakePartialCaseDecision(++i, rootIdentifier, label));
+                    }
                 }
             }
 
             return builder.ToImmutableAndFree();
         }
 
+        private ImmutableArray<PartialCaseDecision> MakeCases(BoundExpression switchGoverningExpression, BoundPattern pattern, LabelSymbol successLabel)
+        {
+            var builder = ArrayBuilder<PartialCaseDecision>.GetInstance(1);
+            var rootIdentifier = new BoundDagTemp(switchGoverningExpression.Syntax, switchGoverningExpression.Type, null, 0);
+            builder.Add(MakePartialCaseDecision(1, rootIdentifier, pattern, successLabel));
+            return builder.ToImmutableAndFree();
```

Comment:
>builder.ToImmutableAndFree(); [](start = 19, length = 29)

Perhaps `return ImmutableArray.Create(MakePartialCaseDecision(...));` rather than using a builder.

Reply:
Yes (will adjust in a new iteration).

---
In reply to: [172597831](https://github.com/dotnet/roslyn/pull/24059#discussion_r172597831) [](ancestors = 172597831)